### PR TITLE
Make EmailSink public

### PR DESCRIPTION
**Pull request by chrisbrain**
This at least allows it to be created with a custom ITextFormatter from a client project which will include all custom properties on the even to also be included in the email.

### CODE REVIEW

**Comment by merbla** [comment]
Hey @chrisbrain thats for the PR.  A couple of things.

Firstly sinks are generally internal and exposed via the fluent config aka [LoggerConfig](https://github.com/serilog/serilog-sinks-email/blob/master/src/Serilog.Sinks.Email/LoggerConfigurationEmailExtensions.cs).  This helps for a few reasons one of those is to ensure the fluent interface stays true to the sink.

Also, we have a few checks on the build so any `public` classes/methods need comments.

Be interested to see your thoughts.

**Comment by chrisbrain** [comment]
Hi,

I did wonder if having sinks internal was a pattern you adopted so I picked one at random, MongoDB, and noted it was public:
https://github.com/serilog/serilog-sinks-mongodb/blob/master/src/Serilog.Sinks.MongoDB/Sinks/MongoDB/MongoDBSink.cs

I just picked another, SignalR, also appears to be public:
https://github.com/serilog/serilog-sinks-signalr/blob/master/src/Serilog.Sinks.SignalR/Sinks/SignalR/SignalRSink.cs

So would a better pull request be one giving an opportunity to override the ITextFormatter via the fluent interface?

Basically I am looking at the most sensible route to being able to have all custom properties from the even included in the email body and the easiest way I can spot is to plug in my own implementation of the ITextFormatter, something I have now done by grabbing all the source for the Email Sink which I did not really want to have to do and it doesn't feel like it should be necessary.

**Comment by merbla** [comment]
@chrisbrain sorry the delay, yes I think a PR with the fluent config would be preferable (bearing in mind backwards compatibility).

Let me know how you go!

**Comment by chrisbrain** [comment]
Just created a new pull request:
https://github.com/serilog/serilog-sinks-email/pull/6
